spec/src/main/asciidoc/metrics_spec.adoc

```diff
@@ -197,6 +197,12 @@ The Metadata:
 * displayName (optional): A human readable name of the metric for display purposes if the metric name is not
 human readable. This could e.g. be the case when the metric name is a uuid.
 * tags (optional): A list of `key=value` pairs, which are separated by comma. See also <<supplying_of_tags>>.
+* reusable (optional): If set to `true`, then it is allowed to register a metric multiple times under the same name.
+Note that all such instances must set `reusable` to `true`.
+Default is `false`.
+The implementation must throw an `IllegalArgumentException` in case an attempt is made to register a metric multiple times where not all instances
+are marked as reusable.
```

Comment:
hmm...do you mean... The implementation must throw an `IllegalArgumentException` in case an attempt is made to register a metric using the same name as a previous registration where any previous registration indicates the metric is not reusable.

Reply:
Yes. Common, you must read between the lines :-)

Reply:
"not all instances are marked as reusable" is worded in a way that sounds like you might be asking if I know what all registrations within my app are going to set for this flag (regardless of whether those parts of the code have attempted their registration yet).

Reply:
It means "you can not mix registration attempts where not all attempts have `reusable=true`".

If the 1st place registers with `reusable=false`, then the metric is not reusable.
If the 1st .. nth place registers with `reusable=true` and the n+1st registers with `reusable=false` then the n+1st has to fail.

Reply:
I think I know what you're trying to say (and it's tricky to say in a short way).  How about this?... The implementation must throw an 'IllegalArgumentException' during a metric registration call when the call would result in the reuse of a metric where that metric was either previously declared not reusable or where the registration call itself declares the metric to not be reusable.